diff --git a/node_modules/@tanstack/router-generator/.bun-tag-deccea1aff07dad2 b/.bun-tag-deccea1aff07dad2
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/cjs/config.cjs b/dist/cjs/config.cjs
index 4f3e9627d87a57e45533007ca0e2efd7a333e61d..1ac9266f9da79f504b86ff302cc0ebf0c6547ba1 100644
--- a/dist/cjs/config.cjs
+++ b/dist/cjs/config.cjs
@@ -31,7 +31,7 @@ const configSchema = baseConfigSchema.extend({
   enableRouteTreeFormatting: zod.z.boolean().optional().default(true),
   routeTreeFileFooter: zod.z.union([
     zod.z.array(zod.z.string()).optional().default([]),
-    zod.z.function().returns(zod.z.array(zod.z.string()))
+    zod.z.function().output(zod.z.array(zod.z.string()))
   ]).optional(),
   autoCodeSplitting: zod.z.boolean().optional(),
   customScaffolding: zod.z.object({
diff --git a/src/config.ts b/src/config.ts
index 5c663a5593c0bd26b1c086b2c400193bf3f1c7e1..db15d7045030d045437fc8910c72e0272bd92228 100644
--- a/src/config.ts
+++ b/src/config.ts
@@ -40,7 +40,7 @@ export const configSchema = baseConfigSchema.extend({
   routeTreeFileFooter: z
     .union([
       z.array(z.string()).optional().default([]),
-      z.function().returns(z.array(z.string())),
+      z.function().output(z.array(z.string())),
     ])
     .optional(),
   autoCodeSplitting: z.boolean().optional(),
